frontend/packages/ceph-storage-plugin/src/components/create-storage-system/create-storage-system-steps/backing-storage-step/backing-storage.tsx

```diff
@@ -0,0 +1,189 @@
+    */
+    if (hasOCS && externalProviders.length) {
+      dispatch({ type: 'backingStorage/setType', payload: BackingStorageType.EXTERNAL });
+    }
+  }, [dispatch, externalProviders.length, hasOCS]);
+
+  const { type, externalProvider } = state.backingStorage;
+  const { storageClass } = state;
+
+  const showExternalSystemSelection = type === BackingStorageType.EXTERNAL;
+  const showStorageClassSelection = !hasOCS && type === BackingStorageType.EXISTING;
+  const RADIO_GROUP_NAME = 'backing-storage-radio-group';
+
+  const onRadioSelect = (_, event) => {
+    dispatch({ type: 'backingStorage/setType', payload: event.target.value });
+    dispatch({
+      type: 'currentStep/resetCount',
+    });
+  };
+
+  return (
+    <ErrorHandler error={error} loaded={loaded}>
+      <Form>
+        <Radio
+          label={t('ceph-storage-plugin~Use an existing storage class')}
+          description={t(
+            'ceph-storage-plugin~Can be used on all platforms except BareMetal. OpenShift Data Foundation will use an infrastructure storage class provided by the hosting platform.',
+          )}
+          name={RADIO_GROUP_NAME}
+          value={BackingStorageType.EXISTING}
+          isChecked={type === BackingStorageType.EXISTING}
+          onChange={onRadioSelect}
+          isDisabled={hasOCS}
+          body={
+            showStorageClassSelection && (
+              <StorageClassSelection dispatch={dispatch} selected={storageClass} />
+            )
+          }
+          id={`bs-${BackingStorageType.EXISTING}`}
+        />
+        <Radio
+          label={t('ceph-storage-plugin~Create a new storage class using local devices')}
+          description={t(
+            'ceph-storage-plugin~Can be used on any platform having nodes with local devices. The infrastructure storage class is provided by Local Storage Operator on top of the local devices.',
+          )}
+          name={RADIO_GROUP_NAME}
+          value={BackingStorageType.LOCAL_DEVICES}
+          isChecked={type === BackingStorageType.LOCAL_DEVICES}
+          onChange={onRadioSelect}
+          isDisabled={hasOCS}
+          id={`bs-${BackingStorageType.LOCAL_DEVICES}`}
+        />
+        <Radio
+          label={t('ceph-storage-plugin~Connect a new external storage system')}
+          description={t(
+            'ceph-storage-plugin~Can be used to connect an external storage platform to OpenShift Data Foundation.',
+          )}
+          name={RADIO_GROUP_NAME}
+          value={BackingStorageType.EXTERNAL}
+          isChecked={type === BackingStorageType.EXTERNAL}
+          onChange={onRadioSelect}
+          isDisabled={externalProviders.length === 0}
+          body={
+            showExternalSystemSelection && (
+              <ExternalSystemSelection
+                selectedProvider={externalProvider}
+                dispatch={dispatch}
+                providersList={externalProviders}
+                t={t}
```

Comment:
```suggestion
```
This doesn't need to be passed as a prop. It can be directly used inside the component.